onnxruntime/core/optimizer/nhwc_transformer.cc

```diff
@@ -237,6 +248,32 @@ void NhwcTransformerImpl::TransformQLinearGlobalAveragePool(Node& node) {
   CreateNhwcArgument(node, node, nhwc_input->rank_);
 }
 
+void NhwcTransformerImpl::TransformSplit(Node& node) {
+  auto& input_defs = node.MutableInputDefs();
+
+  auto* nhwc_input = LookupNhwcArgument(input_defs[0]);
+  if (nhwc_input == nullptr) {
+    return;
+  }
+
+  // Change the axis attribute accordingly for NCHW to NHWC model.
+  const auto* axis_attr = graph_utils::GetNodeAttribute(node, "axis");
+  if (axis_attr != nullptr && utils::HasInt(*axis_attr)) {
+    int axis = static_cast<int>(axis_attr->i());
+    if (axis >= 1 && axis < nhwc_input->rank_) {
+      int new_axis = ((axis == 1) ? static_cast<int>(nhwc_input->rank_) - 1 : axis - 1);
+      node.AddAttribute("axis", static_cast<int64_t>(new_axis));
+    } else if (axis < 0 && axis > -static_cast<int>(nhwc_input->rank_)) {
+      int new_axis = ((axis == -(static_cast<int>(nhwc_input->rank_) - 1)) ? - 1 : axis - 1);
+      node.AddAttribute("axis", static_cast<int64_t>(new_axis));
```

Comment:
This could be refactored that if axis is not in a known range, then return (if axis is 5 and this is a 4D tensor, then don't try to optimize the node). Then fall into a single node.AddAttribute that sets the new axis (instead of the two calls).

Reply:
We should return on invalid axis here.